### DISCUSSION TITLE
Unable to connect using Java

### DISCUSSION THREAD

**Discussion by ghost** [planetscale/discussion#84] [Bug Reports] [answered]
Generated a brand new Spring Boot application with Intellij IDEA and followed the MySQL tutorial from the [official Spring docs](https://spring.io/guides/gs/accessing-data-mysql/).

Also generated a connection string in the PlanetScale dashboard using the Java format.

When I try to run my application I am met with this error:

SQL State  : 08S01
Error Code : 1815
Message    : Can't route the user '__planet_scale_username__'
Caused by : java.sql.SQLNonTransientConnectionException: Can't route the user '__planet_scale_username__'

Relevant Spring Boot configuration settings:

spring.datasource.url=jdbc:mysql://${MYSQL_HOST}:3306/${MYSQL_DATABASE}?sslMode=VERIFY_IDENTITY
spring.datasource.username=${MYSQL_USERNAME}
spring.datasource.password=${MYSQL_PASSWORD}
spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver

**Comment by ghost** [comment]
Perhaps I missed something in the docs and am using PlanetScale in a way that is not supported.

The git-like process is cool but I am perfectly fine using Flyway with Spring to fully manage and update my database schema. It's as simple as running the app and the migrations are magically applied.

Is the only way to use PlanetScale to manually modify the schema via your CLI?

**Comment by ghost** [comment, marked as answer]
Just watched a YouTube video with Nick (VP of Engineering at PlanetScale) and have concluded that I am happy to use the CLI for the entirety of my schema management needs.

Will go ahead and mark this as solved.

**Reply by samlambert** [reply]
Great to hear.
